lua/autorun/server/sv_mcsp.lua

```diff
@@ -0,0 +1,59 @@
+local MCSPmodel = "models/milaco/minecraft_pm/minecraft_pm.mdl"
+
+util.AddNetworkString("MCSPToSVBroadcastNewMaterial")
+util.AddNetworkString("MCSPToSVResetSkin")
+util.AddNetworkString("MCSPToCLUpdateNewMaterial")
+util.AddNetworkString("MCSPToCLUpdateOnFirstConnect")
+util.AddNetworkString("MCSPToCLResetMaterial")
+
+net.Receive("MCSPToSVBroadcastNewMaterial", function()
+	local mcspurl = net.ReadString()
+	local mcspent = net.ReadEntity()
+
+	if (mcspent.SetMCSPTimeout or 0) > CurTime() - 10 then mcspent:Notify("Cooldown...") return end --cooldown
+	mcspent.SetMCSPTimeout = CurTime()
+
+	if mcspent:GetModel() != MCSPmodel then return end
+	if string.len(mcspurl) > 36 then netply:ChatPrint("URL is too long!") return end
```

Comment:
Use the function SanitizeImgurId() here. Takes a string. Returns just the "abcd.png" part of the url or nil if it is bad. The ImgurMaterial thing on clientside expects to get just the "abcd.png" part so they go together